### PR TITLE
Add CacheLatestTtlSecs to allow expiration of latest schemas

### PR DESCRIPTION
**Pull request by rayokota**
Add CacheLatestTtlSecs to allow expiration of latest schemas

### CODE REVIEW

File: schemaregistry/cache/lrucache.go [hunk 1]
```diff
@@ -131,6 +131,19 @@ func (c *LRUCache) Delete(key interface{}) {
 	}
 }
 
+// Clear clears the cache
+func (c *LRUCache) Clear() {
+	c.cacheLock.Lock()
+	for key, value := range c.lruElements {
```
Comment (atxalex8) [inline schemaregistry/cache/lrucache.go:137]: Is there a more efficient way to clear than iterating through all the elements?

File: schemaregistry/schemaregistry_client.go [hunk 2]
```diff
@@ -260,18 +264,34 @@ func NewClient(conf *Config) (Client, error) {
...
+			return nil, err
+		}
 	} else {
 		schemaToIDCache = cache.NewMapCache()
 		idToSchemaCache = cache.NewMapCache()
 		schemaToVersionCache = cache.NewMapCache()
 		versionToSchemaCache = cache.NewMapCache()
+		latestToSchemaCache = cache.NewMapCache()
+	}
+	if conf.CacheLatestTTLSecs > 0 {
```
Comment (atxalex8) [inline schemaregistry/schemaregistry_client.go:278]: I'm not familiar with the use case here, but we can have a LruCache with TTL on individual entries, why do we want to clear the entire cache vs having TTLs on each cache entry

File: schemaregistry/cache/lrucache.go (hunk 1 above)
Comment (rayokota) [reply schemaregistry/cache/lrucache.go:137]: I believe this is idomatic go before go 1.21, where they added a `clear()` method.  In any case, these maps are not expected to get that large as they store one entry per subject

File: schemaregistry/schemaregistry_client.go (hunk 2 above)
Comment (rayokota) [reply schemaregistry/schemaregistry_client.go:278]: The TTL is just to ensure that the latest entries are refreshed every so often.  It's not necessary to have a TTL per entry.

File: schemaregistry/schemaregistry_client.go [hunk 3]
```diff
@@ -260,18 +264,34 @@ func NewClient(conf *Config) (Client, error) {
...
 	}
 	handle := &client{
 		restService:          restService,
 		schemaToIDCache:      schemaToIDCache,
 		idToSchemaCache:      idToSchemaCache,
 		schemaToVersionCache: schemaToVersionCache,
 		versionToSchemaCache: versionToSchemaCache,
+		latestToSchemaCache:  latestToSchemaCache,
 	}
 	return handle, nil
```
Comment (emasab) [inline schemaregistry/schemaregistry_client.go:296]: Coulse use `runtime.SetFinalizer` with `handle` that stops the ticker and a done channel to exit the for loop in the spawned goroutine

Comment (rayokota) [reply schemaregistry/schemaregistry_client.go:296]: Yes, let me make that change

Comment (rayokota) [reply schemaregistry/schemaregistry_client.go:292]: @emasab , I added the finalizer

**Review by emasab — APPROVED** [review]
LGTM. Thanks @rayokota !

**Pull request merged**